internal/k8s/controllers/node_controller.go

```diff
@@ -65,16 +68,25 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 }
 
 func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	p := predicate.NewPredicateFuncs(
-		func(obj client.Object) bool {
-			node, ok := obj.(*v1.Node)
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			newNodeObj, ok := e.ObjectNew.(*corev1.Node)
 			if !ok {
-				level.Error(r.Logger).Log("controller", "NodeReconciler", "error", "object is not node", "name", obj.GetName())
+				level.Error(r.Logger).Log("controller", "NodeReconciler", "error", "new object is not node", "name", newNodeObj.GetName())
+				return true
+			}
+			oldNodeObj, ok := e.ObjectOld.(*corev1.Node)
+			if !ok {
+				level.Error(r.Logger).Log("controller", "NodeReconciler", "error", "old object is not node", "name", oldNodeObj.GetName())
+				return true
+			}
+			// If there is no changes in node labels, ignore event.
+			if labels.Equals(labels.Set(oldNodeObj.Labels), labels.Set(newNodeObj.Labels)) {
 				return false
 			}
-			return node.Name == r.NodeName
-		})
-
+			return oldNodeObj.Name == newNodeObj.Name && newNodeObj.Name == newNodeObj.Name
```

Comment:
I'd keep only the new name equal, trusting the framework that it passed an old and a new instance of the same object.